tools/clang/test/HLSLFileCheck/hlsl/workgraph/noinput_rdat.hlsl

```diff
@@ -0,0 +1,371 @@
+// ==================================================================
+
+// CHECK:DxilRuntimeData (size = {{[0-9]+}} bytes):
+// CHECK:  StringBuffer (size = {{[0-9]+}} bytes)
+// CHECK:  IndexTable (size = {{[0-9]+}} bytes)
+// CHECK:  RawBytes (size = {{[0-9]+}} bytes)
+// CHECK:  RecordTable (stride = {{[0-9]+}} bytes) FunctionTable[4] = {
+// CHECK:   <0:RuntimeDataFunctionInfo{{.*}}> = {
+// CHECK:      Name: "noinput_broadcasting"
+// CHECK:      UnmangledName: "noinput_broadcasting"
+// CHECK:      Resources: <RecordArrayRef<RuntimeDataResourceInfo>[0]> = {}
+// CHECK:      FunctionDependencies: <string[0]> = {}
+// CHECK:      ShaderKind: Node
+// CHECK:      PayloadSizeInBytes: 0
+// CHECK:      AttributeSizeInBytes: 0
+// CHECK:      FeatureInfo1: 0
+// CHECK:      FeatureInfo2: 0
+// CHECK:      ShaderStageFlag: (Node)
+// CHECK:      MinShaderTarget: 0xf0068
+// CHECK:      MinimumExpectedWaveLaneCount: 0
+// CHECK:      MaximumExpectedWaveLaneCount: 0
+// CHECK:      ShaderFlags: 0 (None)
+// CHECK:      Node: <0:NodeShaderInfo> = {
+// CHECK:        LaunchType: Broadcasting
+// CHECK:        GroupSharedBytesUsed: 0
+// CHECK:        Attribs: <{{[0-9]+}}:RecordArrayRef<NodeShaderFuncAttrib>[3]>  = {
+// CHECK:          [0]: <0:NodeShaderFuncAttrib> = {
+// CHECK:            AttribKind: ID
+// CHECK:            ID: <0:NodeID> = {
+// CHECK:              Name: "noinput_broadcasting"
+// CHECK:              Index: 0
+// CHECK:            }
+// CHECK:          }
+// CHECK:          [1]: <1:NodeShaderFuncAttrib> = {
+// CHECK:            AttribKind: NumThreads
+// CHECK:            NumThreads: <0:array[3]> = { 1, 1, 1 }
+// CHECK:          }
+// CHECK:          [2]: <2:NodeShaderFuncAttrib> = {
+// CHECK:            AttribKind: DispatchGrid
+// CHECK:            DispatchGrid: <4:array[3]> = { 4, 1, 1 }
+// CHECK:          }
+// CHECK:        }
+// CHECK:        Outputs: <RecordArrayRef<IONode>[0]> = {}
+// CHECK:        Inputs: <RecordArrayRef<IONode>[0]> = {}
+// CHECK:      }
+// CHECK:    }
+// CHECK:    <1:RuntimeDataFunctionInfo{{.*}}> = {
+// CHECK:      Name: "noinput_coalescing"
+// CHECK:      UnmangledName: "noinput_coalescing"
+// CHECK:      Resources: <RecordArrayRef<RuntimeDataResourceInfo>[0]> = {}
+// CHECK:      FunctionDependencies: <string[0]> = {}
+// CHECK:      ShaderKind: Node
+// CHECK:      PayloadSizeInBytes: 0
+// CHECK:      AttributeSizeInBytes: 0
+// CHECK:      FeatureInfo1: 0
+// CHECK:      FeatureInfo2: 0
+// CHECK:      ShaderStageFlag: (Node)
+// CHECK:      MinShaderTarget: 0xf0068
+// CHECK:      MinimumExpectedWaveLaneCount: 0
+// CHECK:      MaximumExpectedWaveLaneCount: 0
+// CHECK:      ShaderFlags: 0 (None)
+// CHECK:      Node: <1:NodeShaderInfo> = {
+// CHECK:        LaunchType: Coalescing
+// CHECK:        GroupSharedBytesUsed: 0
+// CHECK:        Attribs: <{{[0-9]+}}:RecordArrayRef<NodeShaderFuncAttrib>[2]>  = {
+// CHECK:          [0]: <3:NodeShaderFuncAttrib> = {
```

Comment:
Do we need every line of RDAT dump?  Note that the `3:` here is fragile, since it indicates the index of this attribute in the `NodeShaderFuncAttribTable`, which could change if attribute change for some function in this library.

This recursively dumps the contents of the RecordRef (from other tables), the first time referenced from each top-level record.  So it's redundant to check the contents of those same records in that corresponding table later.

Reply:
I modeled it after empty_broadcasting_nodes.hlsl, which includes every line. I'm not particular though, it was just easier this way.